spark_client/services.py

```diff
@@ -374,18 +374,15 @@ def get_primary(self) -> Optional[ServiceAccount]:
         all_accounts = self.all()
 
         if len(all_accounts) == 0:
-            raise NoAccountFound(
-                "There are no service account available. "
-                "Please create a primary service account first."
-            )
+            self.logger.warning("There are no service account available.")
```

Comment:
Has implications for callers of get_primary() to properly handle the case when 'None' is returned.

To further this endeavour of avoiding ugly stack traces

Reply:
The problem here is that `get_primary` is also used in other methods, like `set_primary`, where the exception is not handled. I noticed this during the tutorial, where I did created a non-primary account, and then tried to set its primary flag, I was getting an error. 

Looking at the signature, I saw it was `Optional[ServiceAccount]` and description was saying that if no primary account was found, it would return just None, not raise an Exception. So, really, I believe this one was a small bug, as now behaviour adhere to description/signature